Initial port to Vue3, vite, Framework7 v7.1.5, pinia stores, codemirror 6, vue-i18n v11, ...

jsjames opened the pull request:
Closes #2151.

Fully functioning release.
There are some things left to do, see #3378.

Please report issues in this GitHub discussion: https://github.com/openhab/openhab-webui/discussions/3381
We will then decided how critical the reported bug issue and depending on that, fix it right now or put it on the to do lost: https://github.com/openhab/openhab-webui/issues/3378

jsjames committed:
Rebase squash - fix lint issues

Merge commits on vue3 branch

Squashed all vue3 commits

scope-css, other bugs

fix theme-filled by applying to html element (similar to theme-dark)

Formatting, address window.getComputedStyle issue.

Vue3Trend

Basic support for Codemirror and addressed Service settings issue

Build scripts now working

fixed page-details, several tabs, ...

Converted primary states store to pinia

Updated userStore to pinia

Converted components store to pinia

Added runtime store in pinia

Moved to Framework7 v7.1.5

Framework 7.1.5
converted additional vuex store to pinia

misc fixes / rebase

Fix auto dark mode, import echart dark theme, misc other fixes.

removed prettier for the time being
updated all f7-tab with proper tab-link syntax
blockly is now coming up

use precompiled nearley grammar

Signed-off-by: Jimmy Tanagra <jcode@tanagra.id.au>

jest -> vitest

Signed-off-by: Jimmy Tanagra <jcode@tanagra.id.au>

Fix model-tree-view selection

Signed-off-by: Jimmy Tanagra <jcode@tanagra.id.au>

lint fixes

Adapt to vue3-clipboard

Signed-off-by: Jimmy Tanagra <jcode@tanagra.id.au>

fix clipboard icons

Signed-off-by: Jimmy Tanagra <jcode@tanagra.id.au>

use @click.stop

Signed-off-by: Jimmy Tanagra <jcode@tanagra.id.au>

Fix minor issues to fix build failure

Signed-off-by: Jimmy Tanagra <jcode@tanagra.id.au>

Fixed rendering issues of vue3-masonry-css to address the layout issues. To fix, i had to add class "masonry-grid-item" to the instantitionof MasonryGridItem AND prevented the inclusion of child items by commenting out code in the findChildNodesWithClassName function in the vue3-masonry-css library

Fixed AddModule for rules

Fixed f7Modal.close()

Addressed start-up issues with getComputedStyle returning error

wrap components inside f7-list inside f7-list-group

f7-list triggers an error when it contains direct component. They must be wrapped inside <ul>, <div>, or <f7-list-group>. The wrapper cannot be inside the component itself.

Signed-off-by: Jimmy Tanagra <jcode@tanagra.id.au>

fix includeItemNames
still bug in item-picker

removed all $set in html code

Convert props from array to object

Signed-off-by: Jimmy Tanagra <jcode@tanagra.id.au>

fix method casing in setup-wizard

Signed-off-by: Jimmy Tanagra <jcode@tanagra.id.au>

Align max-attributes-per-line eslint setting
Additional fixes for codemirror

fix merge issue

tsconfig & typescript lint (still some pending lint issues)

make codemirror usable

fix input/change event
fix up various modes
add indentation markers
fix tab indenting
fix readonly

Signed-off-by: Jimmy Tanagra <jcode@tanagra.id.au>

move mode for reactivity

Signed-off-by: Jimmy Tanagra <jcode@tanagra.id.au>

misc lint fixes

various lint fixes

Develop sidebar fixes

Fixed expression-tester in developer sidebar
Renamed pinia stores to align with convention
Misc lint fixes

fixed width for vue3-trend
misc fixes in typescript config

Fixed f7SmartSelect in item-picker

fix space before parenthesis for function definitions
Fix arrow-parens
revert definition files to main

Thorough file-by-file review compared to main

Cleaned up defineAsyncComponent
Implemented connection-health-mixin with the useStatesStore()
fixed all f7SmartSelect references
fix varScope issue

Fixing tab-link

misc fixes, nearley support for es6 import

fix item-picker setValueText option

misc fixes

numerous fixes, including formatting clean-up for advanced-labels/checkboxes

fix routes.js loadAsync prop function call

Signed-off-by: Jimmy Tanagra <jcode@tanagra.id.au>

fix theme.aurora

Signed-off-by: Jimmy Tanagra <jcode@tanagra.id.au>

fix editor layout

Signed-off-by: Jimmy Tanagra <jcode@tanagra.id.au>

listen on all hosts

Signed-off-by: Jimmy Tanagra <jcode@tanagra.id.au>

fix theme.aurora

Signed-off-by: Jimmy Tanagra <jcode@tanagra.id.au>

cleaned up advanced-label css and reactivity

fix oh-plan-page and icon transparency
fix "Clear" link on several pages

Explicity called out back-links to make more robust
various other fixes

codemirror autocompletion for ui components

Signed-off-by: Jimmy Tanagra <jcode@tanagra.id.au>

Fixed misc issues, denoted TODO with a V3.0 and V3.1

codemirror: add rules editor hint

add YAML Linter
add lint gutter
fix lint tooltip clipping
refactor and DRY common hintXxx functions
add hintParameters

Signed-off-by: Jimmy Tanagra <jcode@tanagra.id.au>

codemirror lint fixes

Signed-off-by: Jimmy Tanagra <jcode@tanagra.id.au>

fix eslint

Signed-off-by: Jimmy Tanagra <jcode@tanagra.id.au>

fix login issues

lint config update

move the f7-toolbar element to the beginning

Signed-off-by: Jimmy Tanagra <jcode@tanagra.id.au>

Converted all the pinia store
Fixed timing hole where get would continuously be called
misc other fixes

i18n language support fixed

remove tern and add a copy of ecmascript.json

Signed-off-by: Jimmy Tanagra <jcode@tanagra.id.au>

codemirror: support javascript completion

Signed-off-by: Jimmy Tanagra <jcode@tanagra.id.au>

i18n improvements / cleanup

codemirror support double quotes `items.getItem("`

Signed-off-by: Jimmy Tanagra <jcode@tanagra.id.au>

Many bug / css fixes

Reverted having to import Dom7 to a global $$
lint clean

misc bug fixes

fix numerous bugs, re-introduce the meta security insertion

fix numerous bugs, re-introduce the meta security insertion

Signed-off-by: Jeff James <jeff@james-online.com>

jsjames committed:
Prep for PR

Signed-off-by: Jeff James <jeff@james-online.com>

jsjames commented:
> @jsjames i'll try running this on my test system as i'm working on other openHAB development stuff and report any issues here as i find them.

Thanks Dan - I have a built version of the UI here as well: https://drive.google.com/file/d/1DUzyYBjmxaCmBWLXIPCHj0p794WqqG-N/view?usp=drive_link

jsjames committed:
added patch-package

florian-h05 committed:
Review changes

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
Fix not reachable screen unlocalized

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
Make sure to always merge locale instead of setting/overwriting

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
Revert some review changes

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
Add package lock file

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 reviewed (commented):
First of all, thousand thanks for taking on this big challenge!!!

Overall, the changes look really good. I have made a few minor changes during review and discovered several bugs which I will work on:

- [x] openHAB not reachable screen not localised -> fallback locale translations need to be loaded early
- [x] i18n fallback locale unavailable if REST API connection fails -> fallback locale should be "hardcoded"
- [x]  i18n translation loading should merge so that already loaded translations are still available.
- [x] Developer Sidebar title not properly aligned
- [x] Help button in upper right corner not highlighted if dev sidebar open & cannot close it again
- [x] Developer Sidebar Help "FAQ" and "How do I" broken
- [x] Developer Sidebar docs rendering on main page gives `[object Object]` back
- [x] Styling of config parameters looks weird, to much left padding
- [x] home page seems to reload on tab change
- [x] sidebar cannot be pinned anymore
- [x] cron expression editor broken
- [x] dirty check broken
- [x] slider is broken

What I haven't reviewed yet are:

- `components/config/controls/script-editor.vue`
- `components/config/editor/hint-*.js`
- `components/config/yaml-utils.js`
- `css/app.styl`
- `TODO.md`
- `vite.config.ts`

To-Dos:

- [ ] Upgrade marked: 13 -> 14 breaks markdown list rendering, to be investigated
- [ ] on initial loading of setup-wizard, the UI is unresponsive. Corrects itself on reloading (I haven't been able to spend time to debug yet, but thought it should be listed).
- [x] Blockly generates JS code with semicolons, causing ESLint (is there now ESLint integrated in the code editor??) errors

florian-h05 committed:
Developer sidebar fixes

- Fix markdown links broken in help sidebar
- Downgrade marked to fix issues with MD list rendering
- Fix developer-dock title not centered
- Fix help-sidebar FAQs & Quick Start broken
- Fix developer-dock-icon.vue not filled & cannot close dock again

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
Remove padding-left for list items in parameter-group

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
item.vue: Revert change where slot was accidentally replaced with template

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
Fix sidebar pin missing

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
Remove slugify package

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
Clean-up transformation EditorModes mapping

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
Add some TODOs for after the upgrade

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
script-editor: Disable ESLint semicolon rules (allow both with and without semis)

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
Fix cronexpression editor doesn't open & title not aligned properly

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
Fix dirty checks not executed

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 reviewed (commented):
I have now looked through all the code.

The following issues remain and need to be addressed before a merge:

- [x] oh-slider is broken (doesn't initialize, doesn't react to input)
- [x] In dark mode, labels in Echarts aren't readable due to dark text color
- [ ] Echarts localisation doesn't work anymore

The following things need do be looked into later:

- [ ] Upgrade marked: 13 -> 14 breaks markdown list rendering, to be investigated
- [ ] On initial loading of setup-wizard, the UI is unresponsive, but a reload fixes this issue

I will also remove the TODO.md file and convert it to an GitHub issue.

florian-h05 committed:
Fix ECharts labels not readable on dark background

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
Fix a minor Vue warning

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
Revert echarts downgrade

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
Fix slider

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
Adjust Aurora theme to be more like Framework7 v5

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
Home: Fix tab switch causes page to rerender

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
Adjust login/profile styling

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
Fix script editor not editable lock pushes editor down

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
Adjust add-ons swiper to Swiper component change

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
Thing details: Handle Promise error

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
Thing inbox: Fix OK -> Edit not working

Signed-off-by: Florian Hotze <dev@florianhotze.com>

jsjames committed:
Potential fix for localization of echarts

jsjames committed:
removed setup function which was included in error in oh-charts-component

florian-h05 committed:
Minor changes to ECharts locale code

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
Always check if OHApp prop is function before calling it

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
generic-widget-component: Fix Content should render without additional container

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
Delete TODO, now tracked as issue

https://github.com/openhab/openhab-webui/issues/3378

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
Implement state tracking for SVG canvas

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
Fix Items vlist height calculation

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
oh-repeater: Fix fragment renders with container & Fix for use inside oh-swiper/f7-swiper

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
oh-swiper: Fix slide not working

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
link-add: Work-around ChannelList import error to avoid functional regressions

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
oh-label-card: Only render media slot if icon set

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
item-details: Fix title not aligned, links not shown

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
Item details: Fix direct group member edit not working

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
sitemap-edit: Fix tab change not working

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
Fix rendering issue when returning to tabbed page from subpage

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
Fix Items & Things list searchbar not focused on desktop

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
Rule edit: Fix lock icon pushes down script editor

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
Script editor: Disable linter if script is read only

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
Parameter-location: Fix map picker not shown

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
Theme adjustments

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
runtime store: Fix docSrcUrl to also handle milestone versions

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
Fix occasions of missing f7router or runtimeStore

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
Thing details: Fix lock icon pushes code editor down

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
Vite: Disable visualizer open & use port 8081 by default

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
Fix RelativeCI webpack bundle analyzer

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
Make sure to generate build-info before building

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
Adjust webpack-analyzer-report-stats package script

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
Enable compression for production build

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
Fix more occasions of missing f7router

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
Set vite package version

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
oh-context: Simplify code, remove wrapper container

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
generic-widget-component: Fix oh-context won't render in prod

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
Break circular dependency between i18n.ts and useRuntimeStore.ts

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
Item edit: Fix onPageBeforeOut not called

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
Fix location picker doesn't display map

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
Fix occasions of missing f7router

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
Config sheet: Fix overflow y of config params

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
Further Aurora theme adjustments

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
oh-map-page: Fix cartocdn url

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
oh-map-page: Fix map init fails

Signed-off-by: Florian Hotze <dev@florianhotze.com>

jsjames committed:
Fixed sitemape-edit unit tests accounting for $refs being empty

florian-h05 committed:
Fix lint

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
Fix navbar title alignment

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
Further Aurora theme adjustments

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
Move leaflet from dev to prod deps

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
Move @iconify/vue from dev to prod deps

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
Add dependency-cruiser for dependency analysis

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
Add developer dock icon where missing

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
Fix navbar title alignment on iOS

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
Fix wrong classnames for some group & Z-Wave network popups

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
Make sure to generate build-info before running tests

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
Simplify navbar changes from this PR

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
Model: Fix cannot edit Item on desktop

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 reviewed (commented):
Just a quick comment:
This starts working really good, apart from the bugs mentioned in #3378.

FYI I added dependency-cruiser as dev dependency to perform some dependency analysis.

There are a few things left to test for me:
- floorplan pages
- map views with markers
- layout: masonry

jsjames committed:
Removed localization for echarts given issues with production build. will revisit after initial merge

florian-h05 committed:
Model card related style adjustments

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
Things list: Refactor "show no location" checkbox

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
Fix for Items/Things back link not working

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
oh-list-item: Fix equipment accordion not working

Signed-off-by: Florian Hotze <dev@florianhotze.com>

jsjames committed:
Implement a global reactive $f7dim object which can be used by components

jsjames committed:
Implements echart localisation loading based on runtime locale

jsjames committed:
Further optimizations for EN and ZH echart locales

jsjames committed:
Fixed oh-trend width issue

florian-h05 committed:
Fix lint

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
Use global reactive $f7dim everywhere instead of f7.width/f7.height

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
Minor build script enhancements

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
Fix item state preview not rendered in item-details page in prod

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
Rewrite developer-dock-icon using TS Composition API

Signed-off-by: Florian Hotze <dev@florianhotze.com>

jsjames committed:
Hack to fix the f7router issues with addLink and editLink

jsjames committed:
Better hack for the f7router issue where allowPageChange is false

jsjames committed:
Fix for model drag-drop - created deep reactive data moveState structure vs. modifiying of props to ensure reactivity

ghys commented:
![itshappening](https://media1.tenor.com/m/9VuG1d2QOOQAAAAd/oh-my-god-its-happening.gif)

florian-h05 committed:
Use oh-nav-content in most f7-navbars

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
oh-nav-content: Minor refactoring

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
Fix searchbar input not respecting dark mode for orange navbar theme

Fixes https://github.com/openhab/openhab-webui/discussions/3381#discussioncomment-14670571.

Signed-off-by: Florian Hotze <dev@florianhotze.com>

jsjames committed:
Fix includeItemName and includeItemTags mapping from useRuntimeStore in model.vue

jsjames committed:
Add includeNonSemantic in model and model-picker-popup in runtimeStore to make persistent

jsjames committed:
moveState propagated via provide/inject instead of props

jsjames committed:
Fixed drag-drop for semantics tags by providing/injecting moveState so that it will be reactive across components.

jsjames committed:
Fix issue with new navbar title being undefined prior to the item load.

florian-h05 committed:
Fix lint

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
Fix missing page title

Signed-off-by: Florian Hotze <dev@florianhotze.com>

jsjames committed:
Changes necessary to get typescript to build without errors.

florian-h05 committed:
Fix unpinned sidebar again visible after reload

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
oh-nav-content: Add large property for large title

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
Settings page: Use normal back link & Fix searchbar positioning

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
About, Add-on Store, Dev Tools: Dynamic instead of hardcoded back link

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
oh-map-page: Revert accidental removal of mapOptions

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
oh-map-page: Fix markers

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
oh-sipclient: Minor code format changes

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
oh-sipclient: Fix session reactivity issues in Vue template

Signed-off-by: Florian Hotze <dev@florianhotze.com>

jsjames committed:
Fix for setup-wizard being unresponsive on initial load (after login)

florian-h05 committed:
Fix wrong path for initial REST request

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
Fix embedded SVG config popup does't open

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
Fix SIP client local settings popup not working

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
Fix lint

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
Model drag & drop: Minor code changes

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
Model drag & drop: Fix move into model not working

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 commented:
@jsjames From my POV, we are finally ready to merge this!
Please let me know what you think!

florian-h05 reviewed (approved):
LGTM now, thousand thanks for the initial work @jsjames and @jimtng as well as the intense work in the last weeks @jsjames.

I think this is now ready to be merged and make its way into the snapshots 🚀

jsjames committed:
Fix for accessing componentsStore arrays as functions instead of properties.

jsjames committed:
Merge branch 'vue3' of https://github.com/jsjames/openhab-webui into vue3

florian-h05 committed:
Fix f7router missing in widget-config-popup

Signed-off-by: Florian Hotze <dev@florianhotze.com>

florian-h05 committed:
Fix vitest

Signed-off-by: Florian Hotze <dev@florianhotze.com>
